homeassistant/components/media_player/philips_js.py

```diff
@@ -0,0 +1,206 @@
+        self._muted = False
+        self._program_name = None
+        self._channel_name = None
+        self._source = None
+        self._source_list = []
+        self._connfail = 0
+        self._source_mapping = {}
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return True
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_PHILIPS_JS
+
+    @property
+    def state(self):
+        """Get the device state. An exception means OFF state."""
+        try:
+            if self._connfail:
+                self._connfail -= 1
+                self._source = None
+                return STATE_OFF
+            requests.get(DEVICE_NAME_URL.format(self._host), timeout=5)
+            self._state = STATE_ON
+            return self._state
+        except requests.exceptions.RequestException:
+            self._connfail = 5
+            return STATE_OFF
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        return self._source
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    def select_source(self, source):
+        """Set the input source."""
+        if source in self._source_mapping:
+            data = dict(id=self._source_mapping[source])
+            try:
+                resp = requests.post(DEVICE_SRC_SET_URL.format(self._host),
+                                     data=json.dumps(data), timeout=5)
+                if resp.status_code == 200:
+                    self._source = source
+            except requests.exceptions.RequestException:
+                _LOGGER.error('Could not set source to %s', source)
+        else:
+            _LOGGER.warning('invalid source: %s', source)
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        if self._volume is not None:
+            return self._volume
+        else:
+            return None
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    def turn_off(self):
+        """Turn off the device."""
+        self.send_key('Standby')
+
+    def volume_up(self):
+        """Send volume up command."""
+        self.send_key('VolumeUp')
+
+    def volume_down(self):
+        """Send volume down command."""
+        self.send_key('VolumeDown')
+
+    def mute_volume(self, mute):
+        """Send mute command."""
+        self.send_key('Mute')
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._source
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest date and update device state."""
+        try:
```

Comment:
I'm aware of that. I did it like this for two reasons: 1. Creating an extra lib for a handful of requests seems like a little overkill. And even more important: 2. The jointSPACE API seems to be a dead project, so I won't expect any changes which would have to be taken care of in this platform. But if you insist, I'll do it.